tests/src/App/DocumentObserver.cpp

```diff
@@ -0,0 +1,426 @@
+    // Base::Interpreter().runString()
+    auto cmd {std::string()};
+
+    // Performing a fusion to create an object that will be added to the App::Part object and
+    // linked inside the App::LinkGroup object
+    Base::Interpreter().runString("from BOPTools import BOPFeatures");
+    Base::Interpreter().runString("bp = BOPFeatures.BOPFeatures(App.activeDocument())");
+    cmd = "bp.make_multi_fuse([\"";
+    cmd += boxName;
+    cmd += "\", \"";
+    cmd += cylName;
+    cmd += "\", ])";
+    Base::Interpreter().runString(cmd.c_str());
+    Base::Interpreter().runString("App.ActiveDocument.recompute()");
+    // The name of the fusion object
+    auto fuseName {_doc->getObject("Fusion")->getNameInDocument()};
+
+    // Adding the fusion to the App::Part object to test the differences between calling
+    // SubObjectT::normalize() with the options argument set to
+    // SubObjectT::NormalizeOption::NoFlatten or set to other values
+    cmd = "App.ActiveDocument.getObject(\"";
+    cmd += partName;
+    cmd += "\").addObject(App.ActiveDocument.getObject(\"";
+    cmd += fuseName;
+    cmd += "\"))";
+    Base::Interpreter().runString(cmd.c_str());
+    Base::Interpreter().runString("App.ActiveDocument.recompute()");
+
+    // Adding the fusion to the App::LinkGroup object to test SubObjectT::normalize() with the
+    // options argument set to SubObjectT::NormalizeOption::ConvertIndex
+    cmd = "App.ActiveDocument.getObject(\"";
+    cmd += lGrpName;
+    cmd += "\").setLink(App.ActiveDocument.getObject(\"";
+    cmd += partName;
+    cmd += "\"))";
+    Base::Interpreter().runString(cmd.c_str());
+    Base::Interpreter().runString("App.ActiveDocument.recompute()");
+
+    // Defining the name of the object that will be used as argument for various calls of the
+    // constructors of App::SubObjectT objects
+    subName = partName;
+    subName += ".";
+    subName += fuseName;
+    subName += ".";
+    subName += boxName;
+    subName += ".Edge1";
+
+    // An empty App::SubObjectT object
+    auto subObjTEmpty {SubObjectT()};
+    // An App::SubObjectT object without sub objects
+    auto subObjTWithoutSubObj {SubObjectT(box, boxName)};
+    // An App::SubObjectT object with sub objects
+    auto subObjTWithSubObj {SubObjectT(lGrp, subName.c_str())};
+    // An App::SubObjectT object used to test SubObjectT::normalize() with the option argument set
+    // to SubObjectT::NormalizeOption::NoElement
+    auto subObjTWithoutEl {SubObjectT(lGrp, subName.c_str())};
+    // An App::SubObjectT object used to test SubObjectT::normalize() with the option argument set
+    // to SubObjectT::NormalizeOption::NoFlatten
+    auto subObjTNoFlatten {SubObjectT(lGrp, subName.c_str())};
+    // An App::SubObjectT object used to test SubObjectT::normalize() with the option argument set
+    // to SubObjectT::NormalizeOption::KeepSubName
+    auto subObjTKeepSubName {SubObjectT(lGrp, subName.replace(0, strlen(partName), "0").c_str())};
```

Comment:
This could be helped with more comments. The partName is replaced with "0"?

Reply:
Hm, is there a mismatch within the test here. Shouldn't this part belong to `subObjTConvertIndex`? Since only booleans are checked, it may be the case that it happens to pass for the wrong test.